scripts/zones/Maze_of_Shakhrami/Zone.lua

```diff
@@ -21,7 +22,13 @@ function onInitialize(zone)
     SetGroundsTome(tomes);
 
     UpdateTreasureSpawnPoint(17588769);
-    
+
+    local whichNM = math.random(0,19);
+    if (whichNM < 10) then
+        SetRespawnTime(Argus, 0, 43200); -- 0-12 hours 
+    else
```

Comment:
You need to use DeterMob here too, false for the one that will spawn, true for the other one.